src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentContext.cs

```diff
@@ -21,23 +22,27 @@ internal class DocumentContext
 
     public DocumentContext(
         Uri uri,
-        IDocumentSnapshot snapshot)
+        IDocumentSnapshot snapshot,
+        VSProjectContext? projectContext)
     {
         Uri = uri;
         Snapshot = snapshot;
+        ProjectContext = projectContext;
     }
 
     public virtual Uri Uri { get; }
 
     public virtual IDocumentSnapshot Snapshot { get; }
 
+    public VSProjectContext? ProjectContext { get; }
+
     public virtual string FilePath => Snapshot.FilePath.AssumeNotNull();
 
     public virtual string FileKind => Snapshot.FileKind.AssumeNotNull();
 
     public virtual IProjectSnapshot Project => Snapshot.Project;
 
-    public virtual TextDocumentIdentifier Identifier => new VersionedTextDocumentIdentifier()
+    public virtual VSTextDocumentIdentifier Identifier => new VSVersionedTextDocumentIdentifier()
     {
         Uri = Uri
```

Comment:
This should specifiy ProjectContext, right?

Reply:
This gets into an interesting question. `VersionedTextIdentifier` inherits from `TextIdentifier` and not `VSTextIdentifier`. Do we add our own here or fix the POCO package?